src/index.ts

```diff
@@ -45,15 +45,26 @@ export interface OptionsSync extends OptionsBase {
 function getDefaultSearchPlaces(name: string): string[] {
     return [
         'package.json',
+        `.${name}rc`,
         `.${name}rc.json`,
+        `.${name}rc.yaml`,
+        `.${name}rc.yml`,
         `.${name}rc.js`,
+        `.${name}rc.ts`,
         `.${name}rc.cjs`,
+        `.${name}rc.mjs`,
         `.config/${name}rc`,
         `.config/${name}rc.json`,
+        `.config/${name}rc.yaml`,
+        `.config/${name}rc.yml`,
         `.config/${name}rc.js`,
+        `.config/${name}rc.ts`,
         `.config/${name}rc.cjs`,
+        `.config/${name}rc.mjs`,
         `${name}.config.js`,
+        `${name}.config.ts`,
         `${name}.config.cjs`,
+        `${name}.config.mjs`,
```

Comment:
This is tricky. As part of this PR it makes sense to only add `.mjs` extension. 

Yaml and ts extensions are not and won't be supported out of the box so they should all be removed. Let's also remove `.${name}rc` as it is an unrelated change. Sorry for the confusion as I meant that only `.mjs` extension should be added in this PR. This PR would also require to add a couple of test cases. One for an `.mjs` file and one for js file with `esm default export`.

We should also consider how cosmiconfig handles esm files when sync api is used. they have a [section](https://www.npmjs.com/package/cosmiconfig#loading-js-modules) about this in the docs. I haven't checked the implementation yet. Can do that in the next day or two.

Reply:
Hi,
thanks, I tried to add some `mjs` tests, but they don't work and I'm not sure what the correct behaviour even is.

Could you please review and advise?